internal/wazeroir/operations.go

```diff
@@ -1027,82 +1084,55 @@ func NewOperationUnreachable() UnionOperation {
+// The engines are expected to branch into OperationBr.Target (U1) label.
+func NewOperationBr(target LabelID) UnionOperation {
+	return UnionOperation{OpKind: OperationKindBr, U1: uint64(target)}
 }
 
-// String implements fmt.Stringer.
-func (o OperationBr) String() string { return fmt.Sprintf("%s %s", o.Kind(), o.Target.String()) }
-
-// Kind implements Operation.Kind
-func (OperationBr) Kind() OperationKind {
-	return OperationKindBr
-}
-
-// OperationBrIf implements Operation.
+// NewOperationBrIf is a constructor for UnionOperation with Kind OperationKindBrIf.
 //
-// The engines are expected to pop a value and branch into OperationBrIf.Then label if the value equals 1.
-// Otherwise, the code branches into OperationBrIf.Else label.
-type OperationBrIf struct {
-	Then, Else BranchTargetDrop
-}
-
-// String implements fmt.Stringer.
-func (o OperationBrIf) String() string { return fmt.Sprintf("%s %s, %s", o.Kind(), o.Then, o.Else) }
-
-// Kind implements Operation.Kind
-func (OperationBrIf) Kind() OperationKind {
-	return OperationKindBrIf
+// The engines are expected to pop a value and branch into OperationBrIf.Then (Us[0]) label if the value equals 1.
+// Otherwise, the code branches into OperationBrIf.Else (Us[1]) label.
+func NewOperationBrIf(thenTarget, elseTarget BranchTargetDrop) UnionOperation {
+	return UnionOperation{
+		OpKind: OperationKindBrIf,
+		Us:     []uint64{uint64(thenTarget.Target), uint64(elseTarget.Target)},
+		Rs:     []*InclusiveRange{thenTarget.ToDrop, elseTarget.ToDrop},
+	}
 }
 
-// OperationBrTable implements Operation.
+// NewOperationBrTable is a constructor for UnionOperation with Kind OperationKindBrTable.
 //
 // This corresponds to wasm.OpcodeBrTableName except that the label
 // here means the wazeroir level, not the ones of Wasm.
 //
-// The engines are expected to do the br_table operation base on the
-// OperationBrTable.Default and OperationBrTable.Targets. More precisely,
-// this pops a value from the stack (called "index") and decide which branch we go into next
-// based on the value.
+// The engines are expected to do the br_table operation base on the default (Us[0], Rs[0]) and
+// targets (Us[1:], Rs[1:]). More precisely, this pops a value from the stack (called "index")
+// and decides which branch we go into next based on the value.
 //
 // For example, assume we have operations like {default: L_DEFAULT, targets: [L0, L1, L2]}.
 // If "index" >= len(defaults), then branch into the L_DEFAULT label.
 // Otherwise, we enter label of targets[index].
-type OperationBrTable struct {
-	Targets []*BranchTargetDrop
-	Default *BranchTargetDrop
-}
-
-// String implements fmt.Stringer.
-func (o OperationBrTable) String() string {
-	targets := make([]string, len(o.Targets))
-	for i, t := range o.Targets {
-		targets[i] = t.String()
+//
+// if targetRanges is nil, it is padded to the length of targetLabels
+func NewOperationBrTable(targetLabels []uint64, targetRanges []*InclusiveRange) UnionOperation {
```

Comment:
```suggestion
func NewOperationBrTable(defalultLabel Label, targetLabels []Labels, defaultRange *InclusiveRange, targetRanges []*InclusiveRange) UnionOperation {
```

Try this change, and avoid the allocation with `append` at the call site of NewOperationBrTable.